components/application-authenticators/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java

```diff
@@ -159,20 +161,19 @@ protected boolean retryAuthenticationEnabled() {
 
     private AuthenticatedUser getUsername(AuthenticationContext context) {
         //username from authentication context.
-        String username = "";
         AuthenticatedUser authenticatedUser = null;
         for (int i = 1; i <= context.getSequenceConfig().getStepMap().size(); i++) {
-            if (context.getSequenceConfig().getStepMap().get(i).getAuthenticatedUser() != null &&
-                    context.getSequenceConfig().getStepMap().get(i).getAuthenticatedAutenticator()
-                            .getApplicationAuthenticator() instanceof LocalApplicationAuthenticator) {
-                authenticatedUser = context.getSequenceConfig().getStepMap().get(i).getAuthenticatedUser();
+            StepConfig stepConfig = context.getSequenceConfig().getStepMap().get(i);
```

Comment:
Couldn't a NPE can occur when either of context/context.getSequenceConfig()/context.getSequenceConfig().getStepMap() null?